merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresPlanRepository.java

```diff
@@ -438,11 +438,13 @@ public void commit() throws NoSuchPlanException {
       try (final var connection = this.dataSource.getConnection()) {
         try (final var updatePlanAction = new UpdatePlanAction(connection)) {
           updatePlanAction.apply(
-              this.planId,
+              this.planId.id(),
               this.name.orElse(null),
               this.startTime.orElse(null),
               this.endTime.orElse(null));
         }
+      } catch (final FailedUpdateException ex) {
+        throw new NoSuchPlanException(this.planId);
       } catch (final SQLException ex) {
         throw new DatabaseException("Failed to update a plan", ex);
```

Comment:
The `FailedUpdateException` tells you that the update was attempted, and SQL executed without issue, but no update actually occurred. I'm unsure if this can actually happen, but the code is written to handle all possible results returned from the PreparedStatement in the action. Basically the SQL returns the number of rows affected by the update, and if that number is not 1 then the update failed to perform what was expected (updating the one relevant row)

The `SQLException` tells you something went wrong during the SQL execution, i.e. the actual database operation itself caused the exception directly.

I suppose the message provided to the `DatabaseException` here makes this confusing, but hopefully that makes it more clear. Let me know what you think!

Reply:
That explanation does help :+1: I think I'm okay with this as-is then. Resolving this comment now.